### ISSUE TITLE
FhirPath update?

### ISSUE THREAD

**Issue by giacomodeliberali** [lhncbc/formbuilder-lhcforms#20]
Hi, I'm having some troubles with the FHIRPath expressions. 

```json
{
  "resourceType": "QuestionnaireResponse",
  "questionnaire": "Questionnaire/0631bd02-e8ac-4f24-9969-aeee03c00988",
  "status": "completed",
  "authored": "2021-07-05T13:09:39.102Z",
  "item": [
    {
      "linkId": "199",
      "text": "Scala condizionale",
      "item": [
        {
          "linkId": "/922",
          "text": "9.2.2",
          "answer": [
            {
              "valueBoolean": true
            }
          ]
        },
        {
          "linkId": "/8511",
          "text": "8.5.11",
          "answer": [
            {
              "valueCoding": {
                "code": "F",
                "display": "F"
              }
            }
          ]
        },
        {
          "linkId": "/192",
          "text": "Parziale condizionale",
          "answer": [
            {
              "valueDecimal": 13
            }
          ]
        }
      ]
    }
  ]
}
```

Using the [online demo tool](https://hl7.github.io/fhirpath.js) i would expect:

- `descendants().item.where(linkId = '/922').answer.value` should equal `true`
- `descendants().item.where(linkId = '/922').answer.valueBoolean` should equal `true`
- `descendants().item.where(linkId = '/8511').answer.value.code` should equal `F`
- `descendants().item.where(linkId = '/8511').answer.valueCoding.code` should equal `F`
- `descendants().item.where(linkId = '/192').answer.value` should equal `13` <-- not working
- `descendants().item.where(linkId = '/192').answer.valueDecimal` should equal `13`

but the builder  expression engine does not support the `.value` (it only supports the typed value, such as `valueCoding`, not anymore in the FhirPath spec). Is there any chance that you can update it to the latest version? Or at least guide me on how to update it locally?

Edit:
Actually expressions not evaluating properly seem to be only the ones which use `valueDecimal`.

![example](https://user-images.githubusercontent.com/4717227/124614760-e08f6a80-de74-11eb-88b9-cdf541a7e7e9.PNG)

**Comment by plynchnlm** [comment]
Your test case examples work fine in the online demo.  The case that is failing is using some other example (the linkIds do not match the example you provided).  Can you provide an updated sample quesitionnaire that shows the issue?  I tried using https://lforms-smart-fhir.nlm.nih.gov/v/r4/fhir/Questionnaire/44249-1-x (which has scored answers), and the expression item.answerOption.extension.where(url='http://hl7.org/fhir/StructureDefinition/ordinalValue').value finds the valueDecimals.

**Comment by giacomodeliberali** [comment]
Using the [provided questionnaire](https://lforms-smart-fhir.nlm.nih.gov/v/r4/fhir/Questionnaire/44249-1-x):

- Update the calculated expression of question 10 to only print the `%q1_value`, in order to quickly check the return value of the next expression
- Replace `q1_value` expression with `%questionnaire.item.where(linkId = '/44250-9').answerOption.where(value.code=%resource.item.where(linkId = '/44250-9').answer.value.code).extension.where(url=%scoreExt).value` (I replaced the `valueCoding` and `valueDecimal` with `value`, but this does not evaluate properly)
- Replacing `value` only for the `valueCoding` seems to work, but not for decimal `%questionnaire.item.where(linkId = '/44250-9').answerOption.where(value.code=%resource.item.where(linkId = '/44250-9').answer.value.code).extension.where(url=%scoreExt).valueDecimal` (this evaluates properly)

The same expression evaluated in the [demo](https://hl7.github.io/fhirpath.js/) `item.where(linkId = '/44250-9').answerOption.where(valueCoding.code='LA6569-3').extension.where(url='http://hl7.org/fhir/StructureDefinition/ordinalValue').value` correctly returns the value decimal 1.

**Comment by giacomodeliberali** [comment]
Hi @plynchnlm , were you be able to reproduce the issue? We continue having some difficulties running the expressions created in the builder in our .NET backend

**Comment by plynchnlm** [comment]
Sorry for the delay.  I've been out.  I have confirmed the problem.  The expression "%questionnaire.item.where(linkId = '/44250-9').answerOption[1].extension[1].value" does not work in LHC-Forms, but does in fhirpath.js (if you define %questionnaire, or replace it with %context).  It looks like it will be fixed by updating fhirpath.js in our "lforms" package, on which formbuilder-lhcforms depends for rendering forms.  (At least, I tried updating lforms, and it fixed the issue there.)  We will try to get this out soon.
